blurring_as_a_service/metrics/metrics_utils.py

```diff
@@ -0,0 +1,142 @@
+from typing import List, Tuple
+
+import matplotlib.pyplot as plt
+import numpy as np
+from PIL import Image
+from PIL.Image import Image as ImageType
+
+
+def process_image_labels(labels):
+
+    true_classes, true_bboxes = parse_labels(labels["true"])
+    pred_classes, pred_bboxes = parse_labels(labels["predicted"])
+
+    tba_true_mask = generate_binary_mask(true_bboxes)
+    tba_pred_mask = generate_binary_mask(pred_bboxes)
+
+    # discard true and pred classes which are licence_plates
+    person_true_bboxes_filtered = [
+        true_bboxes[i] for i in range(len(true_bboxes)) if true_classes[i] == 0
+    ]
+    person_pred_bboxes_filtered = [
+        pred_bboxes[i] for i in range(len(pred_bboxes)) if pred_classes[i] == 0
+    ]
+
+    uba_true_mask = generate_binary_mask(
+        person_true_bboxes_filtered, consider_upper_half=True
+    )
+    uba_pred_mask = generate_binary_mask(
+        person_pred_bboxes_filtered, consider_upper_half=True
+    )
+
+    return tba_true_mask, tba_pred_mask, uba_true_mask, uba_pred_mask
+
+
+def parse_labels(
+    file_path: str,
+) -> Tuple[List[int], List[Tuple[float, float, float, float]]]:
+    """
+     Parses a labels file with the following normalized format: [x_center, y_center, width, height]
+
+    :param file_path: The path to the labels file to be parsed.
+    :return: A tuple with two lists: classes and bounding_boxes
+    """
+    with open(file_path, "r") as f:
+        lines = f.readlines()
+    f.close()
```

Comment:
It's not necessary to close files when you use `with` in python. This is because the with when finishes call the `__exit__()` function of the class. And the open has as exit file close.

Reply:
Ahh, I did not know, thanks!